tripper/triplestore.py

```diff
@@ -698,6 +711,104 @@ def add_mapsTo(
+        sandboxing for security.
+        """
+        if func_iri in self.function_repo and self.function_repo[func_iri]:
+            return self.function_repo[func_iri]
+
+        func_name = self.value(func_iri, OTEIO.hasPythonFunctionName)
+        module_name = self.value(func_iri, OTEIO.hasPythonModuleName)
+        package_name = self.value(func_iri, OTEIO.hasPythonPackageName)
+
+        if not func_name or not module_name:
+            raise CannotGetFunctionError(
+                f"no documentation of how to access function: {func_iri}"
+            )
+
+        # Import module implementing the function
+        try:
+            module = importlib.import_module(module_name, package_name)
+        except ModuleNotFoundError:
+            # If we cannot find the module, try to install the pypi
+            # package and try to import the module again
+            pypi_package = self.value(func_iri, OTEIO.hasPypiPackageName)
+            if not pypi_package:
+                raise CannotGetFunctionError(  # pylint: disable=raise-missing-from
+                    f"PyPI package not documented for function: {func_iri}"
+                )
+
+            try:
+                subprocess.run(  # nosec
+                    args=[
+                        sys.executable,
+                        "-m",
+                        "pip",
+                        "install",
+                        pypi_package,
+                    ],
+                    check=True,
+                )
+            except subprocess.CalledProcessError as exc:
+                raise CannotGetFunctionError(
+                    f"failed installing PyPI package '{pypi_package}'"
+                ) from exc
+
+            try:
+                module = importlib.import_module(module_name, package_name)
+            except ModuleNotFoundError as exc:
+                raise CannotGetFunctionError(
+                    f"failed importing module '{module_name}'"
+                    + f" from '{package_name}'"
+                    if package_name
+                    else ""
+                ) from exc
+
+        func = getattr(module, func_name)
+        self.function_repo[func_iri] = func
+
+        return func
+
+    def eval_function(self, func_iri, args=(), kwargs=None) -> "Any":
+        """Evaluate mapping function and return the result.
+
+        Parameters:
+            func_iri: IRI of the function to be evaluated.
+            args: Sequence of positional arguments passed to the function.
+            kwargs: Mapping of keyword arguments passed to the function.
+
+        Returns:
+            The return value of the function.
+
+        Note:
+            The current implementation does not protect against side
+            effect or malicious code.  Be warned!
+            This may be improved in the future.
+        """
+        func = self._get_function(func_iri)
+        if not kwargs:
+            kwargs = {}
+
+        # FIXME: Add sandboxing
```

Comment:
Is there an issue to address this FIXME?

Reply:
An issue has now been created: https://github.com/EMMC-ASBL/tripper/issues/167